### DISCUSSION TITLE
`not:` pseudoclass support in JIT mode

### DISCUSSION THREAD

**Discussion by Jazzmanpw** [tailwindlabs/tailwindcss#5217] [Ideas]
Are you planning to add the `not:` pseudoclass utility? In my case, I'm making a list of elements with `after:content-[',']`, and I have to add `last-of-type:after:content-['']`, but it would be much more CSS-like (in a good way) to use `not(last-of-type:):after:content-[',']`. I don't know if it's easy to implement, but it feels like a very intuitive thing to be added.

**Comment by lukewarlow** [comment]
Being able to negate selectors and media query variants would be a brilliant addition.

**Comment by lukewarlow** [comment]
https://jsfiddle.net/jdw9rtya/ - I made a POC to see what this could look like in terms of raw CSS classes. Not sure if using `not(blah)` syntax is best because `()` might be better used for variant grouping in future and this might cause a conflict? But then maybe this actually is just a form of grouping already?

Also not sure how easy it would be to generate that output using postcss but presumably it's possible?

**Comment by MichaelAllenWarner** [comment]
@adamwathan Would you be amenable to a PR adding `not-` variants for every pseudo-class? I'm thinking it might be fairly simple to add a few lines [here](https://github.com/tailwindlabs/tailwindcss/blob/master/src/corePlugins.js#L127-L150) to get stuff like `not-last-of-type:` and `group-not-focus-within:` out of the box.
